checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,243 @@
+        -> a.contains(b) ?  a : (b.contains(a) ? b : a + "; " + b));
+
+    mSparkJobResult = mergeStatus.collect();
+
+    // If one executor cannot recognize Alluxio, something wrong has happened
+    boolean canRecognizeClass = true;
+    boolean canRecognizeFS = true;
+    for (Tuple2<Status, String> op : mSparkJobResult) {
+      if (op._1().equals(Status.FAILCLASS)) {
+        canRecognizeClass = false;
+      }
+      if (op._1().equals(Status.FAILFS)) {
+        canRecognizeFS = false;
+      }
+    }
+    return canRecognizeClass ? (canRecognizeFS ? Status.SUCCESS : Status.FAILFS) : Status.FAILCLASS;
+  }
+
+  /**
+   * Checks if this Spark driver or executors can recognize Alluxio classes and filesystem.
+   *
+   * @return Status
+   */
+  private Status performIntegrationChecks() {
+    // Checks if Spark driver or executors can recognize Alluxio classes
+    try {
+      Class.forName("alluxio.AlluxioURI");
+      Class.forName("alluxio.client.file.BaseFileSystem");
+      Class.forName("alluxio.hadoop.AlluxioFileSystem");
+    } catch (ClassNotFoundException e) {
+      LOG.error("Failed to find Alluxio classes on classpath ", e);
+      return Status.FAILCLASS;
+    }
+
+    // Checks if Spark driver or executors can recognize Alluxio filesystem
+    try {
+      FileSystem.getFileSystemClass("alluxio", new Configuration());
+    } catch (Exception e) {
+      LOG.error("Failed to find Alluxio filesystem ", e);
+      return Status.FAILFS;
+    }
+
+    return Status.SUCCESS;
+  }
+
+  /**
+   * Gets the current Spark driver or executor IP address.
+   *
+   * @return the current Spark driver or executor IP address
+   */
+  private String getAddress() {
+    String address;
+    try {
+      address = InetAddress.getLocalHost().getHostAddress();
+    } catch (UnknownHostException e) {
+      LOG.debug("cannot get host address");
+      address = "unknown address";
+    }
+    return address;
+  }
+
+  /**
+   * Prints related Spark and Alluxio configuration information.
+   *
+   * @param conf the current SparkConf
+   */
+  private void printConfigInfo(SparkConf conf) {
+    if (mAlluxioHAMode) {
+      LOG.info("Alluixo is running in high availbility mode.");
+    }
+
+    try {
+      for (Tuple2<Status, String> sjr : mSparkJobResult) {
+        if (sjr._1().equals(Status.FAILCLASS)) {
+          LOG.error("Spark executors of IP addresses: " + sjr._2
+              + " cannot recognize Alluxio classes.");
+        } else if (sjr._1().equals(Status.FAILFS)) {
+          LOG.error("Spark executors of IP addresses: " + sjr._2
+              + " cannot recognize Alluxio filesystem.");
+        } else {
+          LOG.info("Spark executors of IP addresses: " + sjr._2
+              + " can recognize Alluxio filesystem.");
+        }
+      }
+    } catch (NullPointerException e) {
+      // If Spark driver integration checks failed, sSparkJobResult will not be assigned
```

Comment:
better to handle the possible `null` value with an `if` statement so that it's clear from the code which value might be null.